modules/gapi/include/opencv2/gapi/streaming/desync.hpp

```diff
@@ -38,17 +38,18 @@ struct GDesync {
 };
 
 template<typename G>
-G desync(const G &g) {
+G desync(const G &g, bool drop) {
```

Comment:
I'd argue about
1) `drop` being a boolean flag -- it is always confusing when you see `f(x, true)` or `f(y, false)` in the code. Why `true`? Why `false`?
2) `drop` being a parameter to `desync()`. You may know (or not to know) if you want to drop frames at the time (or in the context, where) you construct your graph. Note a G-API code may be not a graph but a snippet which is used to build a larger pipeline. You know for sure if you want to drop frames when you compile and run your pipeline -- at that point, your scenario is complete. Why not make it a compile argument, then?

On (2), maybe a good point is to have multiple `desync()` in the graph and set individual drop/no drop policies to those, but I am not sure if it is the case. The implementation may be the same, but at the API level I'd introduce this parameter to the Operation when we do need multiple Operations in the same Computation with different options.

Reply:
I don't see any scenarios where it could be useful to specify `drop_frames` for certain `desync`. Let it be a compile argument